src/wallet/wallet.cpp

```diff
@@ -4511,7 +4511,7 @@ util::Result<MigrationResult> MigrateLegacyToDescriptor(std::shared_ptr<CWallet>
     }
     res.backup_path = backup_path;
 
-    bool success = false;
+    bool success{true};
```

Comment:
I'm not sure about this change; it could cause bugs if we ever add something in the procedure without updating the value.

Yet, if others are happy with it, it is fine for me.

Reply:
Yeah, I think so too. We could revert it back to initialize it with `false` and set it to `true` in the `else` clause where we do `local_wallet->SetWalletFlag(WALLET_FLAG_DESCRIPTORS);`? (or `success = Assert(local_wallet->IsWalletFlagSet(WALLET_FLAG_DESCRIPTORS));`)

Reply:
This previous suggestion https://github.com/bitcoin/bitcoin/pull/32149#discussion_r2024734496 also contained narrowing the scope of the `success` variable that imho should make it less likely to miss updating this variable.

Reply:
Another alternative could be moving the initialisation of `success` close to the conditional as it's done in `CWallet::MarkReplaced`:

https://github.com/bitcoin/bitcoin/blob/65dcbec75661d1652beb927f03b1feab4fab932e/src/wallet/wallet.cpp#L1012-L1016

Reply:
For simplicity, we could just:
```c++
// Do the migration of keys and scripts for non-blank wallets, and cleanup if it fails
if (HasLegacyRecords()) {
    success = DoMigration(*local_wallet, context, error, res);
} else {
    // Make sure that descriptors flag is actually set
    local_wallet->SetWalletFlag(WALLET_FLAG_DESCRIPTORS);
    success = true;
}
```

Leaving the `success=false` initialization.